src/bom/bitcode.rs

```diff
@@ -619,10 +619,12 @@ fn inject_bitcode(chan : &mut mpsc::Sender<Option<Event>>,
 /// directories with source files that have similar names.
 ///
 /// To avoid collisions, we append a hash to each filename
-fn build_bitcode_tar(bc_target : &OsString,
-                     bc_path : &Path,
-                     hash : &str,
-                     tar_file : &mut tempfile::NamedTempFile) -> anyhow::Result<()> {
+fn build_bitcode_tar<O>(bc_target : &OsString,
+                        bc_path : &Path,
+                        hash : &str,
+                        tar_file : &O) -> anyhow::Result<()>
+where for<'a> &'a O: Write
+{
```

Comment:
I got this to build with a maybe simpler:

```rust
fn build_bitcode_tar(bc_target : &OsString,
                        bc_path : &Path,
                        hash : &str,
                        tar_file : impl Write) -> anyhow::Result<()>
{
```

Would that work for what you have in mind?

Reply:
Oh, I didn't know about that syntax.  That works and is much cleaner!